include/erlcloud_aws.hrl

```diff
@@ -7,6 +7,8 @@
           s3_host="s3.amazonaws.com"::string(),
           s3_port=80::non_neg_integer(),
           s3_follow_redirect=false::boolean(),
+          s3_follow_redirect_count=2::non_neg_integer(),
+          s3_bucket_access_method=vhost::vhost|path,
```

Comment:
Path-style method can't be always used. Please see an example described in comments to ```s3_follow_redirect()```.
As for non-compliant naming there is always a risk but currently I don't know about such kind of issues for other AWS resources.